Doc/c-api/init.rst

```diff
@@ -1112,10 +1112,16 @@ Sub-interpreter support
 
 While in most uses, you will only embed a single Python interpreter, there
 are cases where you need to create several independent interpreters in the
-same process and perhaps even in the same thread.  Sub-interpreters allow
-you to do that.  You can switch between sub-interpreters using the
-:c:func:`PyThreadState_Swap` function.  You can create and destroy them
-using the following functions:
+same process and perhaps even in the same thread.
+
+Before sub-interpreters, there is a main interpreter which is the first one
+created by the CPython runtime on startup when the ``python`` command is
+run. The :c:func:`PyInterpreterState_Main` funtion returns a pointer to this
+main interpreter's state.
```

Comment:
This description of the main interpreter could be a bit more direct but the focus should remain within the context of subinterpreters.  For instance, there's no need here to mention any specific API related to the main interpreter.  Anyway, I've updated the BPO issue with a list of key points to cover.

FWIW, I'm not sure whether it would be better to keep the main-related text small (e.g. one short paragraph) or to make it comprehensive (e.g. across multiple paragraphs).  If we go with that latter then it may make sense to have a separate section nearby for the main interpreter.

Reply:
I think it makes sense to keep things minimal for now, and defer more comprehensive documentation to the resolution of the PEP for a Python level subinterpreter API.